Updating core while it is running causes either SIGSEGV, module not found error, or 100% CPU

If you update Core while it is currently running (as do most people), one of three things seem to occur:

1. bcrypto.node not found, which crashes the process:

```
1|ark-rela | Error: Cannot find module 'bcrypto.node'
1|ark-rela |     at moduleError (/home/ark/.config/yarn/global/node_modules/bcrypto/node_modules/loady/lib/loady.js:154:15)
1|ark-rela |     at resolve (/home/ark/.config/yarn/global/node_modules/bcrypto/node_modules/loady/lib/loady.js:89:9)
1|ark-rela |     at load (/home/ark/.config/yarn/global/node_modules/bcrypto/node_modules/loady/lib/loady.js:99:16)
1|ark-rela |     at Object.<anonymous> (/home/ark/.config/yarn/global/node_modules/bcrypto/lib/native/binding.js:11:33)
1|ark-rela |     at Module._compile (internal/modules/cjs/loader.js:1085:14)
1|ark-rela |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
1|ark-rela |     at Module.load (internal/modules/cjs/loader.js:950:32)
1|ark-rela |     at Function.Module._load (internal/modules/cjs/loader.js:790:12)
1|ark-rela |     at Module.require (internal/modules/cjs/loader.js:974:19)
1|ark-rela |     at Module.Hook._require.Module.require (/home/ark/.config/yarn/global/node_modules/require-in-the-middle/index.js:80:39) {
1|ark-rela |   code: 'MODULE_NOT_FOUND'
1|ark-rela | }
```

2. SIGSEGV forcing the running process to quit:

```
PM2          | App [ark-relay:1] exited with code [0] via signal [SIGSEGV]
PM2          | App [ark-relay:1] starting in -fork mode-
PM2          | App [ark-relay:1] online
```

3. Process eats 100% CPU usage until manually restarted:

If a crypto operation occurs, and neither 1 or 2 are triggered, the Core process will get stuck with 100% CPU usage until manually restarted. This even prevents `pm2 restart` from gracefully restarting since it does not respond to either `SIGINT` or `SIGTERM`, so after a 30 second delay, pm2 resorts to `SIGKILL`.
 
All of these, I presume, are caused by the fact the bcrypto library is deleted/rebuilt as part of the update process so the already running process can't find or use it anymore. But this forces the current running instance of Core to crash, which is not desirable behaviour, especially since this can lead to a race condition where a half-updated Core gets auto-restarted by pm2 if the old process dies (triggering a respawn) before the update is fully complete. Core should be able to handle updates gracefully without crashing the currently running process until the user is ready to restart at the end.

--- a/__tests__/unit/core-cli/services/installer.test.ts
+++ b/__tests__/unit/core-cli/services/installer.test.ts
@@ -31,7 +31,7 @@ describe("Installer.install", () => {
 
         installer.install("@arkecosystem/core");
 
-        expect(spySync).toHaveBeenCalledWith("yarn global add @arkecosystem/core@latest --force", { shell: true });
+        expect(spySync).toHaveBeenCalledWith("rm -f `yarn global dir`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@latest --force", { shell: true });
     });
 
     it("should install specific package when tag is provided", () => {
@@ -44,7 +44,7 @@ describe("Installer.install", () => {
 
         installer.install("@arkecosystem/core", "3.0.0");
 
-        expect(spySync).toHaveBeenCalledWith("yarn global add @arkecosystem/core@3.0.0 --force", { shell: true });
+        expect(spySync).toHaveBeenCalledWith("rm -f `yarn global dir`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@3.0.0 --force", { shell: true });
     });
 
     it("should throw when exit code isn't 0", () => {
@@ -57,7 +57,7 @@ describe("Installer.install", () => {
 
         expect(() => installer.install("@arkecosystem/core")).toThrow("stderr");
 
-        expect(spySync).toHaveBeenCalledWith("yarn global add @arkecosystem/core@latest --force", { shell: true });
+        expect(spySync).toHaveBeenCalledWith("rm -f `yarn global dir`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@latest --force", { shell: true });
     });
 });
 

--- a/__tests__/unit/core-cli/services/installer.test.ts
+++ b/__tests__/unit/core-cli/services/installer.test.ts
@@ -31,7 +31,7 @@ describe("Installer.install", () => {
 
         installer.install("@arkecosystem/core");
 
-        expect(spySync).toHaveBeenCalledWith("rm -f `yarn global dir`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@latest --force", { shell: true });
+        expect(spySync).toHaveBeenCalledWith("rm -f \"`yarn global dir`\"/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@latest --force", { shell: true });
     });
 
     it("should install specific package when tag is provided", () => {
@@ -44,7 +44,7 @@ describe("Installer.install", () => {
 
         installer.install("@arkecosystem/core", "3.0.0");
 
-        expect(spySync).toHaveBeenCalledWith("rm -f `yarn global dir`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@3.0.0 --force", { shell: true });
+        expect(spySync).toHaveBeenCalledWith("rm -f \"`yarn global dir`\"/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@3.0.0 --force", { shell: true });
     });
 
     it("should throw when exit code isn't 0", () => {
@@ -57,7 +57,7 @@ describe("Installer.install", () => {
 
         expect(() => installer.install("@arkecosystem/core")).toThrow("stderr");
 
-        expect(spySync).toHaveBeenCalledWith("rm -f `yarn global dir`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@latest --force", { shell: true });
+        expect(spySync).toHaveBeenCalledWith("rm -f \"`yarn global dir`\"/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add @arkecosystem/core@latest --force", { shell: true });
     });
 });
 

--- a/packages/core-cli/src/services/installer.ts
+++ b/packages/core-cli/src/services/installer.ts
@@ -16,7 +16,7 @@ export class Installer {
     public install(pkg: string, tag: string = "latest"): void {
         this.installPeerDependencies(pkg, tag);
 
-        const { stdout, stderr, exitCode } = sync(`rm -f \`yarn global dir\`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add ${pkg}@${tag} --force`, { shell: true });
+        const { stdout, stderr, exitCode } = sync(`rm -f "\`yarn global dir\`"/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add ${pkg}@${tag} --force`, { shell: true });
 
         if (exitCode !== 0) {
             throw new Error(`"yarn global add ${pkg}@${tag} --force" exited with code ${exitCode}\n${stderr}`);

--- a/packages/core-cli/src/services/installer.ts
+++ b/packages/core-cli/src/services/installer.ts
@@ -16,7 +16,7 @@ export class Installer {
     public install(pkg: string, tag: string = "latest"): void {
         this.installPeerDependencies(pkg, tag);
 
-        const { stdout, stderr, exitCode } = sync(`yarn global add ${pkg}@${tag} --force`, { shell: true });
+        const { stdout, stderr, exitCode } = sync(`rm -f \`yarn global dir\`/node_modules/better-sqlite3/build/Release/better_sqlite3.node && yarn global add ${pkg}@${tag} --force`, { shell: true });
 
         if (exitCode !== 0) {
             throw new Error(`"yarn global add ${pkg}@${tag} --force" exited with code ${exitCode}\n${stderr}`);

